app/src/org/commcare/android/db/legacy/LegacySqlIndexedStorageUtility.java

```diff
@@ -126,16 +123,17 @@ public T getRecordForValues(String[] rawFieldNames, Object[] values) throws NoSu
             throw new InvalidIndexException("Invalid unique column set" + rawFieldNames + ". Multiple records found with value " + values.toString(), rawFieldNames.toString());
         }
         c.moveToFirst();
-        byte[] data = c.getBlob(c.getColumnIndexOrThrow(DbUtil.DATA_COL));
+        byte[] data = c.getBlob(c.getColumnIndexOrThrow(DatabaseHelper.DATA_COL));
+        int dbId = c.getInt(c.getColumnIndexOrThrow(DatabaseHelper.ID_COL));
         c.close();
-        return newObject(data);
+        return newObject(data, dbId);
```

Comment:
I'd prefer we move the newObject code in its old form to this class rather than change how this class functions in general. This was a snapshot of a procedural generation/translation process so it's really sketch for us to make changes to that process over time, and cleaner for the code to represent what the legacy object used to do